### DISCUSSION TITLE
Unable to upload images Error 500

### DISCUSSION THREAD

**Discussion by Eric-CENGN** [netbox-community/netbox-docker#698] [Getting Help] [answered]
Hello all, 

I’m using NetBox-docker, and I’m having issues with image uploads to sites/racks/etc. After upgrading to v2.11.12 from v2.9.9, I always get “upstream connect error or disconnect/reset before headers. reset reason: connection termination” or the very descriptive “Error 500.“. We are still able to see/delete previously uploaded images. Has anyone encountered this before?

I couldn’t find anything online outside of the ones specifying permission issues, and I’ve confirmed that permissions are open for the media directory, and NetBox uses root user so I don’t believe the issue is related to that.

On a separate but likely related note, I noticed that our implementation has the media volume mounted to /etc/netbox/media instead of the usual /opt/netbox/netbox/media, and trying to move its mountpoint to /opt/netbox/netbox/media breaks the previously uploaded images, and we are still unable to upload images. Where is the media directory path specified? Is it in the NetBox DB somewhere?

**Comment by kkthxbye-code** [comment]
The media folder is set her:

https://github.com/netbox-community/netbox-docker/blob/release/env/netbox.env#L19

Check the logs of the netbox container, you should be able to find help in the wiki.

https://github.com/netbox-community/netbox-docker/wiki/Troubleshooting

**Reply by Eric-CENGN** [reply]
I've played with setting the media folder back to /opt/netbox/netbox/media, but this has not changed much. It allowed me to have access to already uploaded images when they're mounted to /opt/netbox/netbox/media, but I am still getting the same two errors when trying to upload an image.

On a separate note, I do now get the following on the logs (IP Address and FQDN changed for privacy reasons): 
2022/02/11 21:45:50 [alert] 62#69 *267 mkstemp(/opt/unit/tmp//req-XXXXXXXX) failed (13: Permission denied)
10.0.0.1 - - [11/Feb/2022:21:45:50 +0000] "POST /dcim/sites/6/images/add/ HTTP/1.1" 500 54 "https://netbox.test.com/dcim/sites/6/images/add/" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.15; rv:94.0) Gecko/20100101 Firefox/94.0"

The permission error is confounding to me as the media directory's permissions are as open as can be (chmod 777), and NetBox uses root user.

The Troubleshooting section did not have much helpful information on this subject.

**Comment by Eric-CENGN** [comment, marked as answer]
Issue was caused by what was described here https://github.com/netbox-community/netbox-docker/issues/506 and here https://github.com/netbox-community/netbox-docker/pull/509. Tl;dr, issue is caused by nginx request being too big and attempts to dump it into /opt/unit/tmp/ temporary directory, but "unit" user being unable to modify contents of that directory, as kubernetes initializes the container and, consequently, that directory, using root:root, and so this causes a permission issue. This issue was fixed in a more recent version of netbox-docker, but since we are running an earlier version of it for now, I have fixed it by having the pod run a small script modifying the /opt/unit/tmp/ owners to unit:root at pod startup.
